### PR TITLE
fix(ios): add Apple privacy manifest

### LINKED ISSUE #1622: PRIVACY MANIFEST FILE FOR IOS

### PR DESCRIPTION
**Pull request by RhyG**
## Description

Added a privacy manifest file to comply with Apple's upcoming [privacy manifest requirements](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files). 

Changes were based off Apple's documentation for [describing use of required reason API](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api).

Added values for the following APIs used by the lib:

- NSFileCreationDate
- NSFileSystemFreeSize
- NSFileSystemSize
- NSUserDefaults

Let me know if there's any changes that need to be made.

Fixes #1622 

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅    |
| Android |    n/a     |
| Windows |    n/a     |

## Checklist

* [x] I have tested this on a device/simulator for each compatible OS
* [ ] I added the documentation in `README.md`
* [ ] I updated the typings files (`privateTypes.ts`, `types.ts`)
* [ ] I added a sample use of the API (`example/App.js`)

### CODE REVIEW

File: RNDeviceInfo.podspec [hunk 1]
```diff
@@ -14,6 +14,7 @@ Pod::Spec.new do |s|
 
   s.source       = { :git => "https://github.com/react-native-device-info/react-native-device-info.git", :tag => "v#{s.version}" }
   s.source_files  = "ios/**/*.{h,m}"
+  s.resource = 'ios/PrivacyInfo.xcprivacy'
```
Comment (TMaszko) [inline RNDeviceInfo.podspec:17]: https://github.com/google/promises/pull/226#discussion_r1447871477 :)

Comment (RhyG) [reply RNDeviceInfo.podspec]: Thanks for that, updated with the change.

**Review by LFMAKER — APPROVED** [review]

**Comment by gajjartejas** [comment]
Hi @RhyG When I try to generate Apple Privacy Reports from `Organizer -> Archives -> Generate Privacy Report` I got this error

```
Missing an expected key: 'NSPrivacyCollectedDataTypes' 
FeelApp.app/RNDeviceInfoPrivacyInfo.bundle/PrivacyInfo.xcprivacy
```

Screenshot 2024-03-29 at 12 46 15 PM

here is similar issue: https://github.com/aws-amplify/amplify-swift/issues/3552

**Comment by TMaszko** [comment]
@gajjartejas We need empty `NSPrivacyCollectedDataTypes` array :)
cc: @RhyG

File: ios/PrivacyInfo.xcprivacy [hunk 2]
```diff
@@ -29,5 +29,7 @@
 			</array>
 		</dict>
 	</array>
+	<key>NSPrivacyCollectedDataTypes</key>
+    <array/>
```
Comment (RhyG) [inline ios/PrivacyInfo.xcprivacy:33]: Not sure why Github is formatting it like this, but was added via Xcode and looks fine in VS Code so should be right.

**Comment by RhyG** [comment]
@gajjartejas @TMaszko updated with an empty `NSPrivacyCollectedDataTypes` array. 🙂

**Review by paddlefish — APPROVED** [review]

File: ios/PrivacyInfo.xcprivacy [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>NSPrivacyAccessedAPITypes</key>
+	<array>
```
Comment (c100k) [inline ios/PrivacyInfo.xcprivacy:6]: Should `NSPrivacyAccessedAPICategorySystemBootTime` be added as well ?

Comment (larigeniem) [reply ios/PrivacyInfo.xcprivacy:6]: Yeah need NSPrivacyAccessedAPICategorySystemBootTime aswell

Comment (RhyG) [reply ios/PrivacyInfo.xcprivacy:6]: Done 🙂

**Review by LFMAKER — APPROVED** [review]

**Review by ujeon — APPROVED** [review]

**Comment by cakasuma** [comment]
do we need to put this into our react native privacy manifest file, or merging this will automatically save us from the review?

**Review by TMaszko — APPROVED** [review]

**Review by mwmcode — APPROVED** [review]

**Comment by TMaszko** [comment]
@tdraper-dev That's exactly what I've done in my app :). I defined all APIs that my SDKs could use and I'll be removing those entries once libraries are fully migrated :)

**Comment by CaptainJeff** [comment]
Anyone know what, if anything, this PR is missing in order for it to be merged?

**Review by schie — APPROVED** [review]
Looks good to me. 

@mikehardy any reservations?

If not, we can ship it.

**Comment by schie** [comment]
:shipit:

**Review by leethree — COMMENTED** [review]
@dependabot squash and merge

**Comment by c100k** [comment]
> I'm still having the apple warnings after upgrading to the latest version. Is there another step that I have to take inorder for the privacy manifest to be included in the bundle?

The warnings are probably coming from another library that you're using in your project. For example https://github.com/itinance/react-native-fs or any other library.

**Comment by CaptainJeff** [comment]
@c100k sorry two questions

1) Wouldn't it still show in the Privacy report when i do  Organizer -> Archives -> Generate Privacy Report? I don't see it there
2) Sorry for my ignorance on this one but from Apples eyes are these privacy manfiests encapsulated per pod like that? You dont' just declare them once for the app like you would for something in the info.plist file like `NSBluetoothAlwaysUsageDescription` which satisfies the requirement for the whole binary? I would think if you have the privacy manifest for the api usage in quesiton then they would accept it

**Pull request merged**